### ISSUE TITLE
Conversion to Swift 4

### ISSUE THREAD

**Issue by kaifucious** [marmelroy/PhoneNumberKit#163]
Are plans to update this framework to Swift 4 underway?

**Comment by dpopov** [comment]
I added a pull with upgrades to [swift 4](https://github.com/marmelroy/PhoneNumberKit/pull/164)

**Comment by nab0y4enko** [comment]
@marmelroy, merge, please...
![1434596941_632146314](https://user-images.githubusercontent.com/1070269/30523164-30886824-9be5-11e7-8127-0c058fde5248.jpg)

**Comment by gligorkot** [comment]
Can we please get this merged in? Upgrading my project to Swift 4 and this is the only library that's not working yet.

**Comment by marmelroy** [comment]
Sorry for the (mega) delay with this. 

Swift 4 version released (2.0.0).

**Comment by fluidsonic** [comment]
Still not compiling on Swift 4 in Xcode 9:

```
Pods/PhoneNumberKit/PhoneNumberKit/NSRegularExpression+Swift.swift:16:32:
'distance(from:to:)' is unavailable: Any String view index conversion can fail in Swift 4; please unwrap the optional indices
```
